### DISCUSSION TITLE
Loading AWS Secrets Manager secrets from Parameter Store parameters.

### DISCUSSION THREAD

**Discussion by ManasaBusi** [aws/aws-dotnet-extensions-configuration#178] [Q&A] [answered]
Hello,

Its mentioned in the Read me file that its posssible to load AWS Secrets Manager secrets from Parameter Store parameters and below is the code snippet given. Deos this work only if the secret is in key-value pairs format ?Does it work if the secret is a PlainText?

Also, it says that the Key/Value pairs could be retrieved from the **ConfigurationManager** object. Can someone tell me how I can retrieve the secret value that has been read?
```C#
builder.Services.Configure<Settings>(builder.Configuration.GetSection($"common:settings"));

public class Program
{
    public static void Main(string[] args)
    {
        CreateWebHostBuilder(args).Build().Run();
    }

    public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
        WebHost.CreateDefaultBuilder(args)
            .ConfigureAppConfiguration(builder =>
            {
                builder.AddSystemsManager("/aws/reference/secretsmanager/SomeSecret");
            })
            .UseStartup<Startup>();
}
```

**Comment by ashishdhingra** [comment, marked as answer]
@ManasaBusi Good morning. Could you try changing your code to `builder.AddSystemsManager("/aws/reference/secretsmanager/")` and then try reading `SomeSecret` from the configuration. This library requires to specify the parameter prefix and then loads the parameters under that prefix.

Thanks,
Ashish

**Reply by ManasaBusi** [reply]
This works fine, thank you.
